pkg/operator/hive/hive.go

```diff
@@ -211,3 +212,22 @@ func (r *ReconcileHiveConfig) includeAdditionalCAs(hLog log.FieldLogger, h *reso
 
 	return nil
 }
+
+func (r *ReconcileHiveConfig) includeGlobalPullSecret(hLog log.FieldLogger, h *resource.Helper, instance *hivev1.HiveConfig, hiveDeployment *appsv1.Deployment) {
+	if instance.Spec.GlobalPullSecret == nil {
+		hLog.Debug("GlobalPullSecret is not provided in HiveConfig, it will not be deployed")
+		return
+	}
+	globalPullSecretEnvVars := []corev1.EnvVar{
```

Comment:
Change this to a `corev1.EnvVar` instead of a slice. There is no need to build a slice here.

Reply:
good catch